client-v2/src/main/java/com/clickhouse/client/api/data_formats/internal/AbstractBinaryFormatReader.java

```diff
@@ -275,20 +279,19 @@ public String getString(String colName) {
             return null;
         } else if (value instanceof String) {
             return (String) value;
+        } else {
+            ClickHouseDataType dataType = schema.getColumnByName(colName).getDataType();
+            if (dataType == ClickHouseDataType.Enum8 || dataType == ClickHouseDataType.Enum16) {
+                ClickHouseEnum clickHouseEnum = schema.getColumnByName(colName).getEnumConstants();

```

Comment:
Are we always schema-aware?

Reply:
Yes. Otherwise we may not read data. 
Schema may be from data stream header or user provided.